jupyterhub/app.py

```diff
@@ -1226,39 +1224,49 @@ def user_stopped(user, server_name):
             yield self.proxy.delete_user(user, server_name)
             yield user.stop(server_name)
 
+        @gen.coroutine
+        def check_spawner(user, name, spawner):
+            status = 0
+            if spawner.server:
+                try:
+                    status = yield spawner.poll()
```

Comment:
in KubeSpawner, poll doesn't actually have to be a coroutine. Is there something we can do that optimizes based on that info?

Reply:
If poll isn't a coroutine, making these concurrent will have no benefit whatsoever, so it's better for it to be a coroutine if it's going to take any time at all.